projectile.el

```diff
@@ -2776,6 +2776,142 @@ test/impl/other files as below:
   :type 'function
   :package-version '(projectile . "1.0.0"))
 
+;;;; Constant signifying opting out of CMake preset commands.
+(defconst projectile--cmake-no-preset "*no preset*")
+
+(defun projectile--cmake-version()
+  "Compute CMake version."
+  (let* ((string (shell-command-to-string "cmake --version"))
+         (match (string-match "^cmake version \\(.*\\)$" string)))
+    (when match
+      (version-to-list (match-string 1 string)))))
+
+(defun projectile--cmake-check-version(version)
+  "Check if CMake version is at least VERSION."
+  (and
+   (version-list-<= version (projectile--cmake-version))))
+
+(defconst projectile--cmake-command-presets-minimum-version-alist
+    '((:configure-command . (3 19))
+      (:compile-command . (3 20))
+      (:test-command . (3 20))))
+(defun projectile--cmake-command-presets-supported(command-type)
```

Comment:
I normally add blank lines around function defs.

Reply:
Btw, isn't it easier to just set 3.20 as the minimum version for everything instead of having it so granular?

Reply:
Fixed the blank lines.

I think the combination of how nice presets are, and CMake often being installed via the system package manager on non-rolling distros justifies the level of granularity. The configuration-preset is by far the most valuable. See f.ex: https://packages.debian.org/sid/cmake

I can remove it if it's too much, but I would be hard pressed to make a choice between setting minimum version to 3.19 and dropping support for compile and test-commands or setting the minimum version to 3.20!